corehq/apps/export/models/new.py

```diff
@@ -3091,3 +3091,28 @@ def get_ledger_section_entry_combinations(domain):
 
 # Used to identify tables in a bulk case export
 ALL_CASE_TYPE_TABLE = PathNode(name=ALL_CASE_TYPE_EXPORT)
+
+
+def get_datasource_export_from_config(config):
+    from corehq.apps.userreports.util import get_indicator_adapter
+    adapter = get_indicator_adapter(config)
+    table = adapter.get_table()
+
+    def get_export_column(column):
+        return ExportColumn(
+            label=column.id,
+            item=ExportItem(
+                path=[PathNode(name=column.id)],
+                label=column.id,
+                datatype=column.datatype,
+            ),
+            selected=True,
+        )
+
+    return TableConfiguration(
+        label=table.name,
+        columns=[
+            get_export_column(col)
+            for _, col in config.columns_by_id.items()
```

Comment:
Nit: If you're not interested in the keys, you could use `config.columns_by_id.values()`.